Modules/Tracker/QuestieTracker.lua

```diff
@@ -30,14 +30,89 @@ local lineIndex = 0
 local lastAQW = GetTime()
 local durabilityInitialPosition = nil
+local LSM30 = LibStub("LibSharedMedia-3.0", true)
 
 -- used for fading the background of the tracker
 _QuestieTracker.FadeTickerValue = 0
 _QuestieTracker.FadeTickerDirection = false -- true to fade in
 _QuestieTracker.IsFirstRun = true -- bad code
 
 -- Forward declaration
-local _OnClick, _OnEnter, _OnLeave, _AQW_Insert, _RemoveQuestWatch
+local _OnClick, _OnEnter, _OnLeave, _AQW_Insert, _RemoveQuestWatch, _PlayerPosition, _QuestProximityTimer
+
+local function getWorldPlayerPosition() -- Turns coords into 'world' coords so it can be compared with any coords in another zone
+    local uiMapID = C_Map.GetBestMapForUnit("player");
+    if not uiMapID then
+        return nil;
+    end
+    local mapPosition = C_Map.GetPlayerMapPosition(uiMapID, "player");
+    local _, worldPosition = C_Map.GetWorldPosFromMapPos(uiMapID, mapPosition);
+    return worldPosition;
+end
+
+local function getDistance(x1, y1, x2, y2) -- Basic proximity distance calculation to compare two locs (normally player position and provided loc)
+    return math.sqrt( (x2-x1)^2 + (y2-y1)^2 );
+end
+
+function GetDistanceToClosestObjective(questID) -- main function for proximity sorting
+    local player = getWorldPlayerPosition();
+    if not player then
+        return nil;
+    end
+    local coordinates = {};
+    local quest = QuestieDB:GetQuest(questID);
+    if not quest then return end;
+
+    local spawn, zone, name = QuestieMap:GetNearestQuestSpawn(quest)
+    if not spawn then return end;
+    if not zone then return end;
+    if not name then return end;
+    local _, worldPosition = C_Map.GetWorldPosFromMapPos(ZoneDataAreaIDToUiMapID[zone], {
+        x = spawn[1] / 100,
+        y = spawn[2] / 100
+        });
+    tinsert(coordinates, {
+        x = worldPosition.x,
+        y = worldPosition.y
+        });
+
+    if not coordinates then return end
+    local closestDistance;
+    for _, coords in pairs(coordinates) do
+        local distance = getDistance(player.x, player.y, worldPosition.x, worldPosition.y);
+        if closestDistance == nil or distance < closestDistance then
+            closestDistance = distance;
+        end
+    end
+    return closestDistance;
+end
+
+function getContinent(uiMapID)
+    if not uiMapID then return end;
+    if (uiMapID == 947) or (uiMapID == 1459) or (uiMapID == 1460) or (uiMapID == 1461) then
+        return "Azeroth"
+    elseif ((uiMapID >= 1415) and (uiMapID <= 1437)) or (uiMapID == 1453) or (uiMapID == 1455) or (uiMapID == 1458) or (uiMapID == 1463) then
+        return "Eastern Kingdoms"
+    elseif ((uiMapID >= 1411) and (uiMapID <= 1414)) or ((uiMapID >= 1438) and (uiMapID <= 1452)) or (uiMapID == 1454) or (uiMapID == 1456) or (uiMapID == 1457) then
+        return "Kalimdor"
+    else
+        print(uiMapID, "is unknown")
+    end
+end
+
+function QuestieTracker:updateQuestProximityTimer() -- Check location often and update if you've moved
```

Comment:
Function names should always be in uppercase:

```suggestion
--- Check location often and update if you've moved
function QuestieTracker:UpdateQuestProximityTimer()
```